server/src/main/scala/org/http4s/server/middleware/authentication/BasicAuthentication.scala

```diff
@@ -14,22 +14,56 @@ import scalaz.concurrent.Task
  *              appropriate password.
  */
 object basicAuth {
+  /**
+    * Construct authentication middleware that can validate the client-provided
+    * plaintext password against something else (like a stored, hashed password).
+    * @param realm
+    * @param validate
+    * @return
+    */
+  def apply(realm: String, validate: ValidatePassword): AuthMiddleware[(String, String)] = {
+    challenged(Service.lift { req =>
+      getChallenge(realm, validate, req, validatePassword _)
+    })
+  }
+
+  /**
+    * Construct authentication middleware that compares the client-provided password
+    * to a plaintext password stored on the server.
```

Comment:
Storing it this way is bad practice.  If someone really wants to do this, they can write a non-hashing `validatePassword`.  I'd be okay with eliminating this.